Video player fixes, refs #12902

replaceafill opened the pull request:
This PR:

- Sets metadata preloading of the new `video` tag to scale the MedialElement player better https://github.com/artefactual/atom/commit/1e1a6f1507f12d573f21d4b7e5c95873934d537c
- Fixes audio/video DO loading from the full-width treeview https://github.com/artefactual/atom/commit/b7aee74d45f972e204dde747bef543d668117fc3
- Fixes video thumbnail generation https://github.com/artefactual/atom/commit/a9199fb211fd4862ad361cb0dc1025b58a9c6f47

mcantelon reviewed (approved):
Looks good to me! A ", refs #" should probably be added to the commit message header before merging.

replaceafill committed:
Video player fixes, refs #12902

* Preload video metadata

This helps the MediaElement player to get the aspect ratio before the
user clicks on the play button scaling it better.

* Fix audio/video DO loading from the full-width treeview

This loads the mediaelement assets in the index of the information
object so that the player works when the DOs are loaded through AJAX
calls from interacting with the full-width treeview.

It also converts the mediaelementplayer setup into a Drupal Behavior
(like the previous flowplayer was) so that it can be reattached when
the DOM is updated.

* Fix video thumbnail generation

This fixes the generation of thumbnails for videos shorter than 30
seconds by calculating the half of the video using ffprobe.

It also unifies the thumbnail generation logic with the one used by
the Flash uploader.
